plutus-core/src/Language/PlutusCore/Parsable.hs

```diff
@@ -23,14 +23,14 @@ import           Data.Char       (ord)
 import qualified Data.Text       as T
 import           Text.Read       (readMaybe)
 
-
-class Parsable a
-  where
-  parseConstant :: T.Text -> Maybe a
-  -- ^ Return Nothing if the string is invalid, otherwise the corresponding value of type a.
-  default parseConstant :: Read a => T.Text -> Maybe a
-  parseConstant = readMaybe . T.unpack
-  -- ^ The default implmentation is 'read'
+-- | A class for things that are parsable. Those include tags for built-in types and constants of
+-- such types.
+class Parsable a where
+    parse :: T.Text -> Maybe a
+    -- ^ Return Nothing if the string is invalid, otherwise the corresponding value of type a.
+    default parse :: Read a => T.Text -> Maybe a
```

Comment:
Do we actually want this? :thinking:

Reply:
You mean the default implementation? It's used for everything but `ByteString`, so it's useful.